packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -408,8 +411,12 @@ const DataGridProVirtualScroller = React.forwardRef<
         {leftRenderContext && (
           <VirtualScrollerPinnedColumns
             ref={leftColumns}
-            className={classes.leftPinnedColumns}
-            ownerState={{ side: GridPinnedPosition.left }}
+            className={
+              theme.direction === 'ltr' ? classes.leftPinnedColumns : classes.rightPinnedColumns
+            }
```

Comment:
okay, i will swap `leftPinnedColumns` and `rightPinnedColumns` for `rtl`

Reply:
should I rename the button label "Pin to right" to "Pin to left" when `direction===rtl`? its the easiest solution

Reply:
Why not to set it as in all other areas? Like Start/End? For LTR will be start/end=left/right, for RTL will be start/end=right/left.

Reply:
@MBilalShafi could you confirm that the pinned columns should not be swapped in RTL?

Reply:
> Why swapping the position of the pinned columns? Pin to left is still left in RTL

When using right-to-left (RTL) layout, pinning columns will invert their direction. This means that when attempting to pin a column to the left, it is pinned to the right instead.

LTR
![image](https://user-images.githubusercontent.com/34584266/214496440-ade23244-161f-4569-90b5-fe8bd282ebc6.png)

RTL
![image](https://user-images.githubusercontent.com/34584266/214496299-085d9495-d165-42d0-ba78-b9e7ad71690b.png)

Reply:
This behavior (swapping the pinning direction) is not something I expect as an RTL user.

Even being an RTL user the expectation for `right` would be right side and vice versa, so it doesn't make sense to me to click `Pin to Left` but the column actually pinning to the right side.

Obviously, this way it won't make sense with the demos that are meant for LTR language (English let's say) but that's something expected as the content is LTR.

To improve the RTL user's experience, we can swap the order of column menu items for pinning so that Pin to right comes in start and Pin to left after it because the right should be the first choice for a RTL user when pinning columns just like the left is for LTR user.

Reply:
> To improve the RTL user's experience, we can swap the order of column menu items for pinning so that Pin to right comes in start and Pin to left after it because the right should be the first choice for an RTL user when pinning columns just like the left is for LTR user.

you mean renaming the `Pin to Right` to `Pin to Left`, right? I have tried it before and it works for the users who are pinning the column from the column menu but if you pin the column from the code it will still be in a different direction.

Reply:
> you mean renaming the Pin to Right to Pin to Left, right?

If I understand correctly, it's only about changing the order of the items in the column menu when in the RTL mode.

In LTR mode:
- Pin to left
- Pin to right
	
In RTL mode:
- Pin to right
- Pin to left

Also, the pinned columns shouldn't be swapped in RTL.
In the example below, the `Name` column should be on the left side of the grid and the actions column should be on the right side (same as in LTR):

<img width="846" alt="Screenshot 2023-01-30 at 13 32 06" src="https://user-images.githubusercontent.com/13808724/215477851-da1fd46c-2898-4f01-a997-506cf468efef.png">

Reply:
>Also, the pinned columns shouldn't be swapped in RTL.

this is the default behavior. I have added the inversion before, but @MBilalShafi suggested(https://github.com/mui/mui-x/pull/6580#issuecomment-1404575347) to remove it. I now believe that there may have been a misunderstanding.

>If I understand correctly, it's only about changing the order of the items in the column menu when in the RTL mode.

Oh, I understand now. But, for it to work, I need to add column pinning inversion. Otherwise, the `Pin to Left` option will always be at the beginning.

Reply:
@cherniavskii @MBilalShafi I fixed the column pinning inversion and swapped the order of the column menu pinning item.

Reply:
Thanks for updating the PR @yaredtsy 🙏 
It looks better now. This thread could be resolved now.